Data.cc

```diff
@@ -32,10 +36,70 @@ Data::Data(const std::string& filename) {
     m_data.push_back(entries);
   }
 
+  for (int i = 0; i < size; ++i) {
+    double errors;
+    file >> errors;
+    m_error.push_back(errors);
+  }
+
   // done! close the file
   file.close();
 
   assertSizes();
+
 };
 
+int Data::checkCompatibility(const Data& in, int n) {
+    using namespace std;
+    int count = 0;
+    for (int i=0; i<in.size(); i++){
+        double a = m_data[i] - in.measurement(i);
+        double b = 2*sqrt(pow(m_error[i], 2) + pow(in.error(i), 2));
+        double c = b - a;
+
+        if (c < 0) {
+        count++;
+        }
+    }
+    cout << "number of incompatibilities: " << count << endl;
+    if (count > 0) {
+        cout << "datasets are not compatible" << endl;
+    }
+    else {
+        cout << "datasets are compatible" << endl;
+    }
+    return count;
+}
+
```

Comment:
There are two errors in this function (which causes it to produce wrong results).

First: the integer "n" that you pass to this function is not used at all. It should be used instead of the factor 2 in line 57.
Second: you should use the absolute difference in line 56.